app/models/registry.rb

```diff
@@ -106,14 +106,14 @@ def reachable?
   protected
 
   # Fetch the tag being pushed through the given target object.
-  def get_tag_from_target(target)
+  def get_tag_from_target(namespace, target)
     case target["mediaType"]
     when "application/vnd.docker.distribution.manifest.v1+json",
       "application/vnd.docker.distribution.manifest.v1+prettyjws"
       get_tag_from_manifest(target)
     when "application/vnd.docker.distribution.manifest.v2+json",
       "application/vnd.docker.distribution.manifest.list.v2+json"
-      get_tag_from_list(target["repository"])
+      get_tag_from_list(namespace, target["repository"])
```

Comment:
target["repository"] contains the namespace as a prefix, this will no work in `get_tag_from_list` where `Repository.find_by(name: repository, namespace: namespace)` is called